libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1566,12 +1567,23 @@ string IRGeneratorForStatements::binaryOperation(
 			case Token::BitAnd:
 				fun = "and";
 				break;
+			case Token::SHL:
+				fun = "shl";
```

Comment:
I think shifts should have their own function, you need to revert on negative amount, for example.

Reply:
Yes, likely it will be too complex. Noted signs in the comment above.